classes/galley/DAO.inc.php

```diff
@@ -0,0 +1,308 @@
+            'SELECT COUNT(*) AS row_count
+			FROM publication_galley_settings pgs
+				INNER JOIN publication_galleys pg ON pgs.galley_id = pg.galley_id
+				INNER JOIN publications p ON pg.publication_id = p.publication_id
+				INNER JOIN submissions s ON p.submission_id = s.submission_id
+			WHERE pgs.setting_name = ? AND pgs.setting_value = ? AND pgs.galley_id <> ? AND s.context_id = ?',
+            [
+                'pub-id::' . $pubIdType,
+                $pubId,
+                (int) $excludePubObjectId,
+                (int) $contextId
+            ]
+        );
+        $row = $result->current();
+        return $row ? (bool) $row->row_count : false;
+    }
+
+    /**
+     * @copydoc PKPPubIdPluginDAO::changePubId()
+     */
+    public function changePubId($pubObjectId, $pubIdType, $pubId)
+    {
+        DB::table('publication_galley_settings')
+            ->where('setting_name', 'pub-id::' . $pubIdType)
+            ->where('galley_id', (int) $pubObjectId)
+            ->update(['setting_value' => (string) $pubId]);
+    }
+
+    /**
+     * @copydoc PKPPubIdPluginDAO::deletePubId()
+     */
+    public function deletePubId($pubObjectId, $pubIdType)
+    {
+        $settingName = 'pub-id::' . $pubIdType;
+        $this->deprecatedDao->update(
+            'DELETE FROM publication_galley_settings WHERE setting_name = ? AND galley_id = ?',
+            [
+                $settingName,
+                (int)$pubObjectId
+            ]
+        );
+        $this->deprecatedDao->flushCache();
+    }
+
+    /**
+     * @copydoc PKPPubIdPluginDAO::deleteAllPubIds()
+     */
+    public function deleteAllPubIds($contextId, $pubIdType)
+    {
+        $settingName = 'pub-id::' . $pubIdType;
+
+        $galleyIds = Repo::galley()->getIds(
+            Repo::galley()
+                ->getCollector()
+                ->filterByContextIds([(int) $contextId])
+        );
+
+        foreach ($galleyIds as $galleyId) {
+            $this->deprecatedDao->update(
+                'DELETE FROM publication_galley_settings WHERE setting_name = ? AND galley_id = ?',
+                [$settingName, $galleyId]
+            );
+        }
+        $this->deprecatedDao->flushCache();
+    }
+
+    /**
+     * Get all published submission galleys (eventually with a pubId assigned and) matching the specified settings.
+     *
+     * @param int $contextId optional
+     * @param string $pubIdType
+     * @param string $title optional
+     * @param string $author optional
+     * @param int $issueId optional
+     * @param string $pubIdSettingName optional
+     * (e.g. medra::status or medra::registeredDoi)
+     * @param string $pubIdSettingValue optional
+     * @param DBResultRange $rangeInfo optional
+     *
+     * @return DAOResultFactory
+     */
+    public function getExportable($contextId, $pubIdType = null, $title = null, $author = null, $issueId = null, $pubIdSettingName = null, $pubIdSettingValue = null, $rangeInfo = null)
```

Comment:
I don't love introducing `issueId` to pkp-lib here, especially where it means long lists of parameters in OPS and OMP. It's probably more than you'd like to refactor at the moment, but maybe at least it's a place we could start making use of named arguments.

Reply:
Yeah I don't want to touch these `getExportable()` functions if I can help it. It gets pretty messy trying to restructure these without rewriting the grids wholesale. In this case, the function is used by the pub id plugins. We'd have to rewrite the whole method in the query builder or duplicate all of the code across OJS and OPS.

In my view, it's not worth it now, when our eventual goal is to unify how galleys are handled, refactor the pub id plugins, and remove grids altogether.

Reply:
OK to call it out of scope for now. But could you flag it as deprecated?